lading/src/blackhole/datadog.rs

```diff
@@ -0,0 +1,364 @@
+    };
+
+    counter!("bytes_received", labels).increment(whole_body.len() as u64);
+
+    let content_type = headers
+        .get(header::CONTENT_TYPE)
+        .and_then(|v| v.to_str().ok())
+        .unwrap_or("unspecified");
+    let content_encoding = headers
+        .get(header::CONTENT_ENCODING)
+        .and_then(|v| v.to_str().ok())
+        .unwrap_or("unspecified");
+
+    debug!(
+        path = ?path,
+        content_type = ?content_type,
+        content_encoding = ?content_encoding,
+        body_size = ?whole_body.len(),
+        "received request",
+    );
+
+    let status = match (path.as_str(), content_type) {
+        ("/api/v2/series", "application/x-protobuf") => {
+            handle_v2_protobuf(&whole_body, content_encoding, &path, labels).await
+        }
+        _ => StatusCode::ACCEPTED,
+    };
+
+    Response::builder()
+        .status(status)
+        .body(Full::new(Bytes::new()))
+}
+
+fn decompress_if_needed<'a>(
+    body: &'a [u8],
+    content_encoding: &str,
+) -> Result<Cow<'a, [u8]>, io::Error> {
+    match content_encoding {
+        "gzip" => {
+            let mut decoder = GzDecoder::new(body);
+            let mut decompressed = Vec::new();
+            io::Read::read_to_end(&mut decoder, &mut decompressed)?;
+            Ok(Cow::Owned(decompressed))
+        }
+        "deflate" => {
+            let mut decoder = ZlibDecoder::new(body);
+            let mut decompressed = Vec::new();
+            io::Read::read_to_end(&mut decoder, &mut decompressed)?;
+            Ok(Cow::Owned(decompressed))
+        }
+        "zstd" => {
+            let decompressed = zstd::decode_all(body)
+                .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))?;
+            Ok(Cow::Owned(decompressed))
+        }
+        _ => {
+            // No compression or unknown, avoid allocation
+            Ok(Cow::Borrowed(body))
+        }
+    }
+}
+
+#[inline]
+fn unix_to_instant(timestamp_secs: i64) -> Instant {
+    let now_system = SystemTime::now();
+    let now_instant = Instant::now();
+
+    #[allow(clippy::cast_sign_loss)]
+    let point_time = UNIX_EPOCH + Duration::from_secs(timestamp_secs.max(0) as u64);
+
+    match now_system.duration_since(point_time) {
+        Ok(duration_ago) => now_instant.checked_sub(duration_ago).unwrap_or(now_instant),
+        Err(_) => now_instant,
+    }
+}
+
+async fn handle_v2_protobuf(
+    body: &[u8],
+    content_encoding: &str,
+    path: &str,
+    labels: &[(String, String)],
+) -> StatusCode {
+    let decompressed = match decompress_if_needed(body, content_encoding) {
+        Ok(data) => data,
+        Err(e) => {
+            warn!(
+                "Failed to decompress body for path={path}, encoding={content_encoding}, body_size={}: {e}",
+                body.len()
+            );
+            return StatusCode::BAD_REQUEST;
+        }
+    };
+
+    match MetricPayload::decode(&decompressed[..]) {
+        Ok(payload) => {
+            trace!(
```

Comment:
```suggestion
            debug!(
```
not verbose enough to be trace IMO